lib/actions-oclif/device/move.ts

```diff
@@ -82,47 +83,51 @@ export default class DeviceMoveCmd extends Command {
 		options.application = options.application || options.app;
 		delete options.app;
 
-		const device: ExtendedDevice = await balena.models.device.get(
-			params.uuid,
-			expandForAppName,
-		);
-
-		const belongsToApplication = device.belongs_to__application as Application[];
-		device.application_name = belongsToApplication?.[0]
-			? belongsToApplication[0].app_name
-			: 'N/a';
-
-		// Get destination application
 		let application;
-		if (options.application) {
-			application = options.application;
-		} else {
-			const [deviceDeviceType, deviceTypes] = await Promise.all([
-				balena.models.device.getManifestBySlug(device.device_type),
-				balena.models.config.getDeviceTypes(),
-			]);
-
-			const compatibleDeviceTypes = deviceTypes.filter(
-				(dt) =>
-					balena.models.os.isArchitectureCompatibleWith(
-						deviceDeviceType.arch,
-						dt.arch,
-					) &&
-					!!dt.isDependent === !!deviceDeviceType.isDependent &&
-					dt.state !== 'DISCONTINUED',
+		for (const uuid of params.uuid.split(',')) {
+			const device: ExtendedDevice = await balena.models.device.get(
```

Comment:
If I'm reading this right (I may not be!), a network request for `await balena.models.device.get(...)` is made for each `uuid`, but the resulting `device` object is only actually possibly used for the first device, in order to determine the destination application (and even then, only in case `options.application` was not provided). Because ultimately, to actually move the device, all that is needed seems to be uuid string:
```
    await balena.models.device.move(uuid, tryAsInteger(application));
```

So how about the following structure:
* Extract the first `uuid` (first device), before any `for` loop, something like:  
`const uuid = params.uuid.split(',')[0];`
* Then determine the destination application. Only if needed (if `options.application` was not provided), call `await balena.models.device.get()` for that first uuid only.
* Then, knowing the destination application, have a small for loop moving the devices:
```
	for (const uuid of params.uuid.split(',')) {
		await balena.models.device.move(uuid, tryAsInteger(application));
		console.info(`${uuid} was moved to ${application}`);
	}
```

Reply:
The one problem I see with this method is that `await balena.models.device.get(...)` is validating the uuids exist. If one of the uuids is invalid then the operation will stop before anything is moved. Without this the operation could throw an error and stop with half of the devices moved and the second half left as is. Perhaps the solution for this would be report the uuid that failed to the user and continue trying to move the remaining uuids.

Reply:
That's a good point and feature: validating that all UUIDs exist before attempting to move any of them. 👍 
In this case, how about:
* Query all UUIDs, perhaps even in parallel with `Promise.all`:  
`const devices = await Promise.all(params.uuid.split(',').map((uuid) => balena.models.device.get(...)));`
* Then determine the destination application, using `devices[0]` as needed (if `options.application` was not provided).
* Then, knowing the destination application, have a small for loop moving the devices:
```
	for (const uuid of params.uuid.split(',')) {
		await balena.models.device.move(uuid, tryAsInteger(application));
		console.info(`${uuid} was moved to ${application}`);
	}
```